torch/testing/_internal/common_methods_invocations.py

```diff
@@ -799,6 +850,38 @@ def sample_inputs(self, device, dtype, requires_grad=False):
                                 dtypes=[torch.cfloat, torch.cdouble])),
                    promotes_integers_to_float=True,
                    handles_complex_extremals=False),
+    OpInfo('linalg.pinv',
+           aten_name='linalg_pinv',
+           op=torch.linalg.pinv,
+           dtypes=floating_and_complex_types(),
+           test_inplace_grad=False,
+           supports_tensor_out=False,
+           sample_inputs_func=sample_inputs_linalg_pinv,
+           decorators=[skipCUDAIfNoMagma, skipCPUIfNoLapack],
+           skips=(
+               # TODO: RuntimeError: svd does not support automatic differentiation for outputs with complex dtype.
+               # TODO: Add complex types here
+               # See https://github.com/pytorch/pytorch/pull/47761
+               SkipInfo('TestGradients', 'test_fn_grad', dtypes=[torch.cdouble]),
+               SkipInfo('TestGradients', 'test_fn_gradgrad', dtypes=[torch.cdouble]),
+               SkipInfo('TestGradients', 'test_method_grad', dtypes=[torch.cdouble]),
+               SkipInfo('TestGradients', 'test_method_gradgrad', dtypes=[torch.cdouble]),
+               SkipInfo('TestCommon', 'test_variant_consistency_eager', dtypes=[torch.cfloat, torch.cdouble]),
+               SkipInfo('TestCommon', 'test_variant_consistency_jit', dtypes=[torch.cfloat, torch.cdouble])),
+           ),
+    HermitianOpInfo('linalg.pinv',
```

Comment:
Creating two OpInfos for the same operation is interesting. 

In this case, is the Hermitian variant of this operation effectively a different operation? Or do you think that the OpInfo pattern is just too limited to express what `pinv` needs to be tested effectively?

Reply:
Well, it's both a different operation (`hermitian` bool determines which codepath to take) and current OpInfo pattern is limited. In general, it would be useful for each `SampleInput`to have pre- and post- processing functions. I could do here similarly to what was done in [SVD PR ](https://github.com/pytorch/pytorch/pull/47761/files#diff-d183f2afc51d6a59bc70094e8f476d2468c45e415500f6eb60abad955e065156R79-R86) but decided not to introduce new changes (apart of `variant_test_name`) but use what's available now.

Reply:
It'd be nice to make the OpInfo pattern more flexible and easier to use. I'd like to understand your proposal better. Maybe an issue elaborating on it or a PR demonstrating it would be helpful?